(#651) Add masking for image urls, thread titles and OP ids.

K1rakishou opened the pull request:
Closes #651

I might've forgotten something, so if you find anything just create a new issue.

K1rakishou committed:
(#651) Add masking for image urls, thread titles and OP ids.

Adamantcheese commented on Kuroba/app/src/main/java/com/github/adamantcheese/chan/utils/StringUtils.java:
```diff
@@ -90,4 +90,31 @@ public static String decodeBase64(String base64Encoded) {
+        return postNoString;
+    }
+
+    public static String maskUrl(@NonNull String url) {
```
This should be named maskImageUrl, because that's what it's doing; if you want to mask general URLs, you might want to use a regex based approach using `(\d+)`, and replacing the matching  group with a censored version.

K1rakishou replied:
Done.

Adamantcheese commented on Kuroba/app/src/main/java/com/github/adamantcheese/chan/utils/StringUtils.java:
```diff
@@ -90,4 +90,31 @@ public static String decodeBase64(String base64Encoded) {
+    }
+
+    public static String maskUrl(@NonNull String url) {
+        if (url.length() < 4) {
```
If the URL is less than 4 characters, it's not a URL. `http://` is the bare minimum you need for a URL, and that's not even counting the actual domain and whatnot. Kinda worthless check.

K1rakishou replied:
It's there just to avoid possible OOBEs. I just don't want to convert it into HttpUrl to check whether it is an actual url or not because that will create unnecessary allocations.

Adamantcheese commented on Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/model/orm/Loadable.java:
```diff
@@ -234,9 +236,10 @@ public int hashCode() {
+        return "Loadable{id=" + id + ", mode=" + mode + ", board='" + boardCode + '\'' +
+                ", no=" + maskPostNo(no) + '\'' + ", listViewIndex=" + listViewIndex +
+                ", listViewTop=" + listViewTop + ", lastViewed=" + lastViewed + ", lastLoaded=" +
+                maskPostNo(lastLoaded) + ", markedNo=" + maskPostNo(markedNo) + ", dirty=" + dirty + '}';
```
If you wouldn't mind, add in the loadable downloading state to this String. Seems it was missed out when thread downloading was added.

K1rakishou replied:
Done.

K1rakishou committed:
(#651) Merge branch 'cheese-multi-feature' into (#651)-do-not-print-in-logs-sensitive-info

# Conflicts:
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/downloader/ChunkDownloader.kt
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/downloader/PartialContentSupportChecker.kt

K1rakishou committed:
(#651) maskUrl -> maskImageUrl

K1rakishou committed:
(#651) Add loadableDownloadingState to Loadable.toString()
